common/processes/platform/tiny_process_library/tiny_process_library.cpp

```diff
@@ -168,6 +168,14 @@ error::Error Process::AsyncWait(events::EventLoop &loop, AsyncWaitHandler handle
 	async_wait_data_->event_loop = &loop;
 	async_wait_data_->handler = handler;
 
+	if (async_wait_data_->process_ended) {
+		// The process has already ended. Schedule the handler immediately.
+		auto async_wait_data = async_wait_data_;
```

Comment:
Hmm, why copy it here, and in the lambda capture?

Reply:
Because it's not possible to capture member variables, and we have to capture by value in order to keep our own, independent pointer (so we can't use `this`).

It looks like it works to make this a reference though, and it can still capture by value below, so I switched to that.